examples/bits.rs

```diff
@@ -38,38 +33,77 @@ fn main() {
     // Compute the boundaries of the part of the screen where the turtle will
     // draw
     let right_edge = BIT_BOX * 4.0;
+
+    // Now let's draw some data on the screen, as individual bits
+
+    // Replace this text with a message of your own to see how it changes on the
+    // screen.
+    //
+    // Try translating the displayed lines to binary numbers and work out their
+    // encoding!
+    let text = "¡Hola, mundo! 🌍🌏🌎";
+
+    // Place the turtle according to the length of the text span
     let top_edge = BIT_HEIGHT * (text.len() as f64 / 2.0);
     // The turtle starts from the top right of the region,
     turtle.go_to((right_edge, top_edge));
     // and walks left
     turtle.set_heading(180.0);
 
-    // Now let’s draw some data on the screen, as individual bits
+    // Rust strings can iterate over their individual characters. This block
+    // loops over characters, collecting their start point in the text so that
+    // we can grab the encoded bytes of each one.
+    let mut row_num = 0;
+    for (char_num, (start, codepoint)) in text.char_indices().enumerate() {
+        println!("Character {}: {}", char_num, codepoint);
+        // Each character has a variable width, so we need to find that.
+        let byte_count = codepoint.len_utf8();
+        // And then collect the bytes of the string that make up the character.
+        let row = &text.as_bytes()[start ..][.. byte_count];
```

Comment:
Does this produce a different result from just doing `[start..byte_count]`?

Reply:
Yes. The equivalent single-index expression is `[start .. start + byte_count]`. The use of `..][..` instead of `start +` is a personal habit; I've seen about an even mix of the two styles in the wild, and I haven't observed any noteworthy difference in behavior.

`[start .. byte_count]` is firmly an error, because `byte_count` is always in the range `1 ..= 4`, but `start` monotonically increases through the buffer. Using it will produce a zero-length slice on the second iteration, and panic on the third.

Reply:
Could you change it to `[start..start+byte_count]`? I think given that even I misunderstood that, someone who isn't used to seeing that will probably misunderstand it too. :joy: